### PR TITLE
Synchronization support

### PR DESCRIPTION
**Pull request by sjkelly**

### CODE REVIEW

**Comment by sjkelly** [comment]
xref driver wip: https://github.com/JuliaComputing/xtrx_julia/tree/sjk/sync2

File: gateware/synchro.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from litex.build.io import DDROutput
+
+from litex.soc.interconnect.csr import *
+
+# Synchro ------------------------------------------------------------------------------------------
+
+class Synchro(Module, AutoCSR):
+    def __init__(self, pads):
+        self.control = CSRStorage(fields=[
+            CSRField("int_source", offset=0, size=4, values=[
```
Comment (staticfloat) [inline gateware/synchro.py:19]: Should this be `in_source`?

File: software/soapysdr-xtrx/XTRXDevice.cpp [hunk 2]
```diff
@@ -102,6 +102,11 @@ SoapyXTRX::SoapyXTRX(const SoapySDR::Kwargs &args)
         0 * (1 << CSR_LMS7002M_CONTROL_TX_RX_LOOPBACK_ENABLE_OFFSET)
     );
 
+    //Enable DMA Synchronizer
+    #ifdef CSR_PCIE_DMA0_SYNCHRONIZER_ENABLE_ADDR
+    litepcie_writel(_fd, CSR_PCIE_DMA0_SYNCHRONIZER_ENABLE_ADDR, 0b10);
+    #endif
```
Comment (staticfloat) [inline software/soapysdr-xtrx/XTRXDevice.cpp:108]: Why the `#ifdef` guard?

**Review by staticfloat — APPROVED** [review]

File: gateware/synchro.py (hunk 1 above)
Comment (staticfloat) [reply gateware/synchro.py:19]: **EDIT**: Okay actually I think this is intentional, it's "internal" source. :P

File: software/soapysdr-xtrx/XTRXDevice.cpp (hunk 2 above)
Comment (sjkelly) [reply software/soapysdr-xtrx/XTRXDevice.cpp:108]: When I was using this branch as a base for the peripheral changes: https://github.com/JuliaComputing/xtrx_julia/pull/104/commits/b6a668ebf61ac8545f44481125efa502b50b49e9

Comment (sjkelly) [reply software/soapysdr-xtrx/XTRXDevice.cpp:108]: Yingbo and Chris Elrod turned me on to Clangd. I will get this in that PR with those lint fixes.

**Pull request merged**